src/test/java/htmlpublisher/workflow/PublishHTMLStepTest.java

```diff
@@ -0,0 +1,118 @@
+import hudson.Extension;
+import hudson.model.AbstractProject;
+import hudson.model.FreeStyleProject;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.Builder;
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.junit.Test;
+import static org.junit.Assert.*;
+import org.junit.Rule;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+public class PublishHTMLStepTest {
+
+    @Rule 
+    public JenkinsRule r = new JenkinsRule();
+
+    @Test 
+    public void configRoundTrip() throws Exception {
+        configRoundTrip("Foo", "archive", "index.html");
+        configRoundTrip("Foo2", "archive/foo/bar", "index2.html");
+        configRoundTrip("Foo", "archive", "**");
+        configRoundTrip("Report", "archive", "index.html", false, false, false);
+        configRoundTrip("Report", "archive", "index.html", true, true, true);
+        configRoundTrip("Report", "archive", "index.html", true, false, false);
+        configRoundTrip("Report", "archive", "index.html", false, true, false);
+        configRoundTrip("Report", "archive", "index.html", false, false, true);
+    }
+
+    private void configRoundTrip(String reportName, String reportDir, String reportFiles) throws Exception {
+        configRoundTrip(new HtmlPublisherTarget(reportName, reportDir, reportFiles, 
+                false, false, false));
+    }
+    
+    private void configRoundTrip(String reportName, String reportDir, String reportFiles, 
+            boolean keepAll, boolean alwaysLinkToLastBuild, boolean allowMissing) throws Exception {
+        configRoundTrip(new HtmlPublisherTarget(reportName, reportDir, reportFiles, 
+                keepAll, alwaysLinkToLastBuild, allowMissing));
+    }
+     
+    private void configRoundTrip(@NonNull HtmlPublisherTarget target) throws Exception {
+        configRoundTrip(new PublishHTMLStep(target));
+    }
+    
+    private void configRoundTrip(@NonNull PublishHTMLStep step) throws Exception {
+        FreeStyleProject p = r.createFreeStyleProject();
+        p.getBuildersList().add(new StepBuilder(step));
+        // workaround for eclipse compiler Ambiguous method call
+        p.save();
+        r.jenkins.reload();
+        
+        FreeStyleProject reloaded = r.jenkins.getItemByFullName(p.getFullName(), FreeStyleProject.class);
+        assertNotNull(reloaded);
+        StepBuilder b = reloaded.getBuildersList().get(StepBuilder.class);
+        assertNotNull(b);
+        Step after = b.s;
+        assertNotNull(after);
+        assertEquals(step.getClass(), after.getClass());
+        assertEquals("Initial and reloaded target configurations differ", step.getTarget(), 
+                ((PublishHTMLStep)after).getTarget());
+    }
+
+    @Restricted(NoExternalUse.class)
+    public static final class StepBuilder extends Builder {
```

Comment:
I've started from it, but got into an issue with pom.xml, because I needed the plugin as both basic and test dependency. It didn't work for me for the reason I didn't completely investigate, hence I used a copy-paste fallback (not a good approach).

A functional test would be a right thing, of course